src/js/components/DSLFormWithExpressionUpdates.js

```diff
@@ -0,0 +1,169 @@
+    });
+  }
+
+  /**
+   * Handle a change to the form
+   */
+  handleFormChange({target}) {
+    let {
+      expression,
+      groupCombiner,
+      itemCombiner,
+      onChange,
+      parts,
+      updatePolicy
+    } = this.props;
+    let value = target.value;
+
+    const name = target.getAttribute('name');
+    if (!name) {
+      return;
+    }
+
+    if (target.type === 'checkbox') {
+      value = target.checked;
+    }
+
+    // Locate the respective DSL part node
+    const partNode = parts[name];
+    if (partNode == null) {
+      return;
+    }
+
+    // Locate all the current tokens of the expression that the filter match
+    const matchingNodes = DSLUtil.findNodesByFilter(expression.ast, partNode);
+
+    switch (partNode.filterType) {
+      //
+      // The way an attribute updates depends on the `dslUpdatePolicy`.
+      //
+      // In case of Checkbox poliy, the attribute is added/removed, but in Radio
+      // policy the previous labels must be removed.
+      //
+      case DSLFilterTypes.ATTRIB:
+
+        // On 'Radio' update policy the new value is replacing any occurence
+        // of all of the specified nodes.
+        if (updatePolicy === DSLUpdatePolicy.Radio) {
+          if (value) {
+            expression = DSLUpdateUtil.applyReplace(
+              expression, matchingNodes, [partNode], groupCombiner, itemCombiner
+            );
+          } else {
+            expression = DSLUpdateUtil.applyDelete(
+              expression, matchingNodes
+            );
+          }
+          break;
+        }
+
+        // On 'Checkbox' policy, we just add/remove the matching AST node
+        if (value) {
+          expression = DSLUpdateUtil.applyAdd(
+            expression, [partNode], groupCombiner, itemCombiner
+          );
+        } else {
+          expression = DSLUpdateUtil.applyDelete(
+            expression, matchingNodes
+          );
+        }
+        break;
+
+      //
+      // The exact attribute updates just replaces the node with a new text
+      //
+      case DSLFilterTypes.EXACT:
+        const newExactNode = new FilterNode(0, 0, DSLFilterTypes.EXACT, {
+          text: value
+        });
+
+        expression = DSLUpdateUtil.applyReplace(
+          expression, matchingNodes, [newExactNode], groupCombiner
+        );
+        break;
+
+      //
+      // The fuzzy attribute replaces as many attributes as in the new text
+      //
+      case DSLFilterTypes.FUZZY:
+        const newFuzzyNodes = value.split(' ').map((text) => {
+          return new FilterNode(0, 0, DSLFilterTypes.FUZZY, {text});
+        });
+
+        // And replace the existing fuzzy nodes
+        expression = DSLUpdateUtil.applyReplace(
+          expression, matchingNodes, newFuzzyNodes, groupCombiner
+        );
+        break;
+    }
+
+    // Callback with the new expression
+    onChange(expression);
+  }
+
+  handleFormSubmit(event) {
+    event.preventDefault();
```

Comment:
Can we have the ability to pass an option `onSubmit` property to this component, so that when the user is writing out the `Has words`, then they can click [enter] and the dropdown can "apply"?